integration-tests/src/tests/network/runner.rs

```diff
@@ -36,12 +35,10 @@ use near_primitives::types::{AccountId, ValidatorId};
 use near_primitives::validator_signer::InMemoryValidatorSigner;
 use near_store::test_utils::create_test_store;
 use near_telemetry::{TelemetryActor, TelemetryConfig};
+use std::pin::Pin;
 
-pub type SharedRunningInfo = Arc<RwLock<RunningInfo>>;
-
-pub type ActionFn = Box<
-    dyn FnMut(SharedRunningInfo, Arc<AtomicBool>, &mut Context<WaitOrTimeoutActor>, Addr<Runner>),
->;
+pub type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + 'a>>;
+pub type ActionFn = Box<dyn for<'a> Fn(&'a mut RunningInfo) -> BoxFuture<'a, anyhow::Result<bool>>>;
```

Comment:
https://doc.rust-lang.org/stable/std/ops/enum.ControlFlow.html might be a better fit than bool here

Reply:
done, the amount of extra parentheses is breathtaking :)